gapic-generator-cloud/test/helpers/method/arg_type_test.rb

```diff
@@ -0,0 +1,212 @@
+# frozen_string_literal: true
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "test_helper"
+require "protobuf/descriptors"
+require "google/gapic/schema"
+require_relative "../../../templates/cloud/helpers/method_helper"
+
+describe MethodHelper, :method_arg_type do
+  include MethodHelper
+
+  def descriptor **args
+    Google::Protobuf::FieldDescriptorProto.new args
+  end
+
+  def field descriptor, address, docs = nil, message = nil, enum = nil
+    Google::Gapic::Schema::Field.new descriptor, address, docs, message, enum
+  end
+
+  it "knows the TYPE_DOUBLE type" do
+    type = Google::Protobuf::FieldDescriptorProto::Type::TYPE_DOUBLE # 1
+    arg = field(
+      descriptor(name: "some_method", number: 2, label: 1, type: type),
+      %w[Some Gem some_method]
+    )
+
+    method_arg_type(arg).must_equal String # Decimal
```

Comment:
Is the `Field#type` not working as expected?

Reply:
`Field#type` works as expected, I was told to keep `method_arg_type` faked and to not implement it fully until we have a need to finish the implementation. I was hoping the unfinished test would push us towards fully specifying how this should behave, but that will have to wait for a future change. The structure for finishing the implementation is mostly outlined in this test, however.

Reply:
Oh, that's fine then. I just wanted to make sure we knew how to get the information about the type.